pkg/lso/localvolumediscovery.go

```diff
@@ -0,0 +1,168 @@
+package lso
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/golang/glog"
+	"github.com/openshift-kni/eco-goinfra/pkg/clients"
+	"github.com/openshift-kni/eco-goinfra/pkg/msg"
+	lsoV1alpha1 "github.com/openshift/local-storage-operator/api/v1alpha1"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metaV1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	goclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// LocalVolumeDiscoveryBuilder provides a struct for localVolumeDiscovery object from the cluster
+// and a localVolumeDiscovery definition.
+type LocalVolumeDiscoveryBuilder struct {
+	// localVolumeDiscovery definition, used to create the localVolumeDiscovery object.
+	Definition *lsoV1alpha1.LocalVolumeDiscovery
+	// Created localVolumeDiscovery object.
+	Object *lsoV1alpha1.LocalVolumeDiscovery
+	// Used in functions that define or mutate localVolumeDiscovery definition. errorMsg is processed
+	// before the localVolumeDiscovery object is created
+	errorMsg string
+	// api client to interact with the cluster.
+	apiClient *clients.Settings
+}
+
+// NewLocalVolumeDiscoveryBuilder creates new instance of LocalVolumeDiscoveryBuilder.
+func NewLocalVolumeDiscoveryBuilder(apiClient *clients.Settings, name, nsname string) *LocalVolumeDiscoveryBuilder {
+	glog.V(100).Infof("Initializing new localVolumeDiscovery structure with the following params: name: "+
+		"%s, namespace: %s", name, nsname)
+
+	builder := LocalVolumeDiscoveryBuilder{
+		apiClient: apiClient,
+		Definition: &lsoV1alpha1.LocalVolumeDiscovery{
+			ObjectMeta: metaV1.ObjectMeta{
+				Name:      name,
+				Namespace: nsname,
+			},
+		},
+	}
+
+	if name == "" {
+		glog.V(100).Infof("The name of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'name' cannot be empty"
+	}
+
+	if nsname == "" {
+		glog.V(100).Infof("The nsname of the localVolumeDiscovery is empty")
+
+		builder.errorMsg = "localVolumeDiscovery 'nsname' cannot be empty"
+	}
+
+	return &builder
+}
+
+// Discover fetches existing localVolumeDiscovery from cluster.
+func (builder *LocalVolumeDiscoveryBuilder) Discover() error {
```

Comment:
Please remove this method. 
Add 2 methods instead.
1 - func (builder *LocalVolumeDiscoveryBuilder) Get() 
2 - func Pull